FirebaseAuth/Sources/MultiFactor/FIRMultiFactorResolver.m

```diff
@@ -55,14 +59,21 @@ - (instancetype)initWithMFAPendingCredential:(NSString *_Nullable)MFAPendingCred
 - (void)resolveSignInWithAssertion:(nonnull FIRMultiFactorAssertion *)assertion
                         completion:(nullable FIRAuthDataResultCallback)completion {
 #if TARGET_OS_IOS
-  FIRPhoneMultiFactorAssertion *phoneAssertion = (FIRPhoneMultiFactorAssertion *)assertion;
-  FIRAuthProtoFinalizeMFAPhoneRequestInfo *finalizeMFAPhoneRequestInfo =
-      [[FIRAuthProtoFinalizeMFAPhoneRequestInfo alloc]
-          initWithSessionInfo:phoneAssertion.authCredential.verificationID
-             verificationCode:phoneAssertion.authCredential.verificationCode];
+  NSObject<FIRAuthProto> *finalizedMFARequestInfo;
+  if ([assertion isKindOfClass:[FIRPhoneMultiFactorAssertion class]]) {
+    FIRPhoneMultiFactorAssertion *phoneAssertion = (FIRPhoneMultiFactorAssertion *)assertion;
+    finalizedMFARequestInfo = [[FIRAuthProtoFinalizeMFAPhoneRequestInfo alloc]
+        initWithSessionInfo:phoneAssertion.authCredential.verificationID
+           verificationCode:phoneAssertion.authCredential.verificationCode];
+  } else {
```

Comment:
I would be safer with`else if (is totp assertion){}` and then check if `finalizedMFARequestInfo` is nil before assigning at L76.